### PR TITLE
MM-54199 -  attachment author name missing on mobile

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
There was a small display issue with the author name in the matterpoll attachment in mobile view posts. This PR makes sure to display the author name either the icon is provided or not.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-54199

#### Checklist
- [X ] Added or updated unit tests (required for all new features)
- [ X] Has UI changes
- [ ] Includes text changes and localization file updates
- [X ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Screenshots
Before:
Screenshot 2023-09-29 at 15 50 11

After:
Screenshot 2023-09-29 at 15 22 04

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/components/post_list/post/body/content/message_attachments/attachment_author.test.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import Preferences from '@constants/preferences';
+import {renderWithIntl} from '@test/intl-test-helper';
+
+import AttachmentAuthor from './attachment_author';
+
+describe('AttachmentAuthor', () => {
+    const baseProps = {
+        link: 'http://linktoatachment.com',
+        name: 'jhondoe',
```
Comment (larkox) [inline app/components/post_list/post/body/content/message_attachments/attachment_author.test.tsx:14]: Typo?
```suggestion
        name: 'johndoe',
```

**Review by larkox — APPROVED** [review]
Great!

File: app/components/post_list/post/body/content/message_attachments/attachment_author.test.tsx (hunk 1 above)
Comment (pvev) [reply app/components/post_list/post/body/content/message_attachments/attachment_author.test.tsx:14]: LOL
Screenshot 2023-09-29 at 15 57 48

Comment (larkox) [reply app/components/post_list/post/body/content/message_attachments/attachment_author.test.tsx:14]: So... not a typo then :rofl:

Comment (pvev) [reply app/components/post_list/post/body/content/message_attachments/attachment_author.test.tsx:14]: I knew I remember pretty well my friends name "Jhon", actually like 10 to 15 "Jhons"

**Review by enahum — APPROVED** [review]

**Comment by pvev** [comment]
@DHaussermann friendly reminder on this one :) . Thanks!

**Comment by DHaussermann** [comment]
@pvev maybe I've misunderstood something here but, I don't see a change. The authors name is still not visible in the post attachment data when I view it with the mobile client. I also used the Matterpoll example.


image

I tested this on Android (iOS testing is blocked for me for reasons unrelated to this PR) as follows:
- On community open the test public channel and created a poll with `/poll`
- Noted the authors' name shown in the attachment when viewed in the webapp
- Downloaded [this build](https://pr-builds.mattermost.com/mattermost-mobile/build-pr-7575-3b19fc5/Mattermost_Beta.apk) for PR from the _Mobile: Test Builds_ channel and side-loaded it onto my Android device
- Launched MM Beta app and connected the app to Community server
- Observed the post created in test channel above
- Created a 2nd poll from the mobile app for fun to see if the post would look any different 
**Observed**: The author is still not shown

I using a Samsung Galaxy S22 running Android 13 with the stock One UI 5.1 launcher.

**Comment by DHaussermann** [comment]
Sorry @pvev, we can ignore the comment above ☝️ 
I don't know how but, I seem to have been having a deployment issue earlier.

After repeating the steps and side-loading the app again I can see your change 🎉 
image

Testing is ongoing - I'm still trying to get my iOS device working with test builds. I will post an update on this shortly.

**Comment by DHaussermann** [comment]
Tested and passed.
Tested with `/poll` as it provides the case where there is an author but no icon 
- I see this working as expected.
- When there is  no `author_icon` the post author is still visible in the post content on mobile
- Tested on Androis and iOS

LGTM!

Thanks @pvev for elping troubleshoot my deployment issue on this one 👍

**Review by DHaussermann — APPROVED** [review]

**Pull request merged**